nixos/modules/services/security/authelia.nix

```diff
@@ -379,6 +363,28 @@ in
               Do not include raw secrets in nix settings.
             '';
           }
+          # Removed options. Cannot use mkRemovedOptionModule (or similar methods) due to errors.
+          # See https://github.com/NixOS/nixpkgs/pull/299309#issuecomment-2038240493
+          {
+            assertion = !((instance.settings.server ? host) || (instance.settings.server ? port));
+            message = ''
+              The options `services.authelia.${name}.settings.server.host' and `services.authelia.${name}.settings.server.port'
+              have been replaced with `services.authelia.${name}.settings.server.address'.
+            '';
+          }
+          {
+            assertion = !(instance.secrets ? oidcIssuerPrivateKeyFile);
+            message = ''
+              The option `services.authelia.${name}.secrets.oidcIssuerPrivateKeyFile' has been removed.
```

Comment:
Right now, this is throwing a way more confusing error message:

`error: The option `services.authelia.instances.<name>.secrets.oidcIssuerPrivateKeyFile' does not exist. Definition values:`

If we cannot use the deprecation facilities, we'll still have to leave the option in, in order for the user to ever see this assertion message.